pkg/registry/rbac/reconciliation/reconcile_rolebindings.go

```diff
@@ -204,7 +204,8 @@ func computeReconciledRoleBinding(existing, expected RoleBinding, removeExtraSub
 	switch {
 	case !removeExtraSubjects && len(result.MissingSubjects) > 0:
 		// add missing subjects in the union case
-		result.RoleBinding.SetSubjects(append(result.RoleBinding.GetSubjects(), result.MissingSubjects...))
+		tmp := append([]rbacv1.Subject{}, result.RoleBinding.GetSubjects()...)
+		result.RoleBinding.SetSubjects(append(tmp, result.MissingSubjects...))
```

Comment:
same question, we already own the rolebinding because are working with a deepcopy (`result.RoleBinding = existing.DeepCopyRoleBinding()`)